Plugin.php

```diff
@@ -366,4 +373,56 @@ public function translateRawPlural($string, $count = 0, $params = [], $locale =
     {
         return Lang::choice(Message::transRaw($string, $params, $locale), $count, $params);
     }
+
+    public function extendWinterSitemap()
+    {
+        Event::listen('winter.sitemap.processMenuItems', function ($item, $url, $theme, $apiResult) {
+            if ($item->type === 'cms-page') {
+                return Classes\MLCmsPage::resolveMenuItem($item, $url, $theme);
+            }
+
+            if (PluginManager::instance()->exists('Winter.Pages')) {
+                if ($item->type === 'static-page') {
+                    return Classes\MLStaticPage::resolveMenuItem($item, $url, $theme);
+                }
+            }
+
+            if (PluginManager::instance()->exists('Winter.blog')) {
+                switch ($item->type) {
+                    case 'blog-category':
+                    case 'all-blog-categories':
+                        return Classes\MLBlogCategoryModel::resolveMenuItem($item, $url, $theme);
+                    case 'blog-post':
+                    case 'category-blog-posts':
+                    case 'all-blog-posts':
+                        return Classes\MLBlogPostModel::resolveMenuItem($item, $url, $theme);
+                }
+            }
+            return false;
+        });
+
+        Event::listen('winter.sitemap.makeUrlSet', function ($definition, $xml, $urlSet) {
+            if (Config::get('app.debug', false)) {
+                // hack to force browser to properly render the XML sitemap
+                $nsUrl = 'xmlns:xhtml-namespace-definition-URL-here';
+            } else {
+                $nsUrl = 'http://www.w3.org/1999/xhtml';
+            }
+            $urlSet->setAttribute('xmlns:xhtml', $nsUrl);
+        });
```

Comment:
This should just be done by the Winter.Sitemap plugin itself.

Reply:
Why? This is only needed if xhtml:link nodes are addded (which is not the case in the base plugin)

Reply:
Do you have a link to an article or something that covers the reason for this hack?

Reply:
No, it's just that when there are `xhtml:link` nodes, you need to add the namespace declaration for them, but chrome has problem formatting the xml so it ends up like this:

![Screenshot from 2023-02-28 21-34-24](https://user-images.githubusercontent.com/2013630/222030455-dac675a0-f6cd-4545-b3da-96fab9b7ab7c.png)

instead of like this (with the hack):

![image](https://user-images.githubusercontent.com/2013630/222030585-a6d08976-ebde-4f6c-92b8-ae01173cbc13.png)

Reply:
Can confirm that the hack does what it needs to do.
Readable for us, and readable for bots (I used SiteGuru and it can succesfuly read my sitemap)